### PR TITLE
fix(@desktop/chat): messages sometimes assigned to the wrong user

### PR DESCRIPTION
**Pull request by osmaczko**
dogfooding_1 branch was never merged to master even though it had some important fixes

original PR (done by @saledjenic): https://github.com/status-im/status-desktop/pull/6050

### CODE REVIEW

File: src/app/modules/shared_models/message_model.nim [hunk 1]
```diff
@@ -186,11 +194,18 @@ QtObject:
     of ModelRole.MentionedUsersPks:
       result = newQVariant(item.mentionedUsersPks.join(" "))
 
+  proc updateItemAtIndex(self: Model, index: int) =
+    let ind = self.createIndex(index, 0, nil)
```
Comment (caybro) [inline src/app/modules/shared_models/message_model.nim:198]: Not knowing much (yet) about NIM, but in C++ you'd rather call:

`const auto ind = this.index(index /*row*/, 0 /*col*);`

In other words, I don't see the reason for creating a new index, it should rather call this method: https://doc.qt.io/qt-5/qabstractlistmodel.html#index

File: src/app/modules/shared_models/message_model.nim [hunk 2]
```diff
@@ -186,11 +194,18 @@ QtObject:
...
   proc findIndexForMessageId*(self: Model, messageId: string): int =
+    result = -1
+    if messageId.len == 0:
+      return
     for i in 0 ..< self.items.len:
       if(self.items[i].id == messageId):
-        return i
-    return -1
+        result = i
+        return
```
Comment (caybro) [inline src/app/modules/shared_models/message_model.nim:208]: Shouldn't this method return an `int`? Sorry, still a nim newb but how does this even compile? :)

**Review by caybro — CHANGES_REQUESTED** [review]

File: src/app/modules/shared_models/message_model.nim (hunk 2 above)
Comment (jrainville) [reply src/app/modules/shared_models/message_model.nim:208]: Nim is weird. You can both do `return X` or `result = X` for a return value.
In this case, Patryk set the default return with `result = -1`

Comment (jrainville) [inline src/app/modules/shared_models/message_model.nim:208]: One less line 😄 
```suggestion
        return i
```

**Review by jrainville — APPROVED** [review]

**Comment by iurimatias** [comment]
it's best to rename this to the original PR

File: src/app/modules/shared_models/message_model.nim (hunk 1 above)
Comment (osmaczko) [reply src/app/modules/shared_models/message_model.nim:198]: Anyway, I consider it not to be a blocker of this PR. We should possibly substitute `createIndex` with `index` in follow-up PR.

Comment (caybro) [reply src/app/modules/shared_models/message_model.nim:198]: Sure, let's potentially fix that in some followup PR

File: src/app/modules/shared_models/message_model.nim [hunk 3]
```diff
@@ -120,6 +124,10 @@ QtObject:
     case enumRole:
     of ModelRole.Id:
       result = newQVariant(item.id)
+    of ModelRole.PrevMsgIndex:
+      result = newQVariant(index.row + 1)
+    of ModelRole.NextMsgIndex:
+      result = newQVariant(index.row - 1)
```
Comment (caybro) [inline src/app/modules/shared_models/message_model.nim:130]: Shouldn't "next" be a `+1` and "prev" a `-1`?

**Review by caybro — CHANGES_REQUESTED** [review]

File: src/app/modules/shared_models/message_model.nim (hunk 3 above)
Comment (endulab) [reply src/app/modules/shared_models/message_model.nim:130]: This is a good question :) @caybro there is a comment in ChatMessagesView about it.

**Review by endulab — APPROVED** [review]

File: src/app/modules/shared_models/message_model.nim (hunk 3 above)
Comment (caybro) [reply src/app/modules/shared_models/message_model.nim:130]: Ah, I see, the messages are in reverse order (newest at the bottom)

**Review by caybro — APPROVED** [review]

**Pull request merged**